### PR TITLE
Fix pyyaml/cython issue

### PR DESCRIPTION
**Pull request by maxi297**
## What
Fix following https://github.com/yaml/pyyaml/issues/601

## How
Since only version >6.0.1 has been fixed, we will update pyyaml version to this version

### CODE REVIEW

File: airbyte-cdk/python/Dockerfile [hunk 1]
```diff
@@ -10,6 +10,10 @@ RUN apk --no-cache upgrade \
     && apk --no-cache add tzdata build-base
 
 # install airbyte-cdk
+# FIXME to enable CI to pass, we added this line which allows airbyte-cdk==0.44.4 to work with pyyaml~=5.4
+# We should remove this like as soon as the new version of airbyte-cdk is released. For more information, see
+# https://github.com/yaml/pyyaml/issues/601
+RUN pip install --prefix=/install "Cython<3.0" "pyyaml~=5.4" --no-build-isolation
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-cdk/python/Dockerfile:16]: Can it succeed in the CI and fail on release? Isn't the CI building this image?

**Comment by alafanechere** [comment]
☝️ test on a connector using PyArrow as @pedroslopez mentionned incompatibilities of PyYaml 6 with Pyarrow

**Comment by maxi297** [comment]
> mentionned incompatibilities of PyYaml 6 with Pyarrow

Can you provide more information as to what is incompatible? My best guess right now is that there will be no solution except for waiting for https://github.com/yaml/pyyaml/issues/601#issuecomment-1639873900

**Review by alafanechere — APPROVED** [review]

**Comment by maxi297** [comment]
Perfect! I'll merge as is, release, fix the issue mentioned [here](https://github.com/airbytehq/airbyte/pull/28393#discussion_r1266777719) and re-release

File: airbyte-cdk/python/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxi297) [reply airbyte-cdk/python/Dockerfile:16]: It does only with a hardcoded version of airbyte-cdk so until it is released, I don't think we can fix this. I think this is something we should try to fix eventually...

**Pull request merged**